src/util/fastrange.h

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2018-2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_UTIL_FASTRANGE_H
+#define BITCOIN_UTIL_FASTRANGE_H
+
+#include <cstdint>
+
+// See: https://lemire.me/blog/2016/06/27/a-fast-alternative-to-the-modulo-reduction/
+
+/** Map a value x that is uniformly distributed in the range [0, 2^64) to a
+ *  value uniformly distributed in [0, n) by returning the upper 64 bits of
+ * x * n. */
```

Comment:
A minor nit. Feel free to ignore.

How about putting the reference link after the explanation?

```suggestion
/** Map a value x that is uniformly distributed in the range [0, 2^64) to a
 *  value uniformly distributed in [0, n) by returning the upper 64 bits of
 * x * n. */
 
 // See: https://lemire.me/blog/2016/06/27/a-fast-alternative-to-the-modulo-reduction/
```

As this convention is followed throughout the codebase and seems more logical.

Reply:
I've rewritten the comments in the last commit.